packages/next-app/src/hooks/useTransaction.ts

```diff
@@ -343,6 +383,24 @@ export function useInternalBalancesTransaction({
     }
   }, [submitData]);
 
+  useEffect(() => {
+    if (!error) return;
+    if (
+      operationKind === UserBalanceOpKind.DEPOSIT_INTERNAL &&
+      transactionStatus === TransactionStatus.SUBMITTING
+    ) {
+      setNotification(
+        NOTIFICATION_MAP_INTERNAL_BALANCES[TransactionStatus.WRITE_ERROR]
+      );
+      setTransactionStatus(TransactionStatus.CONFIRMING);
+    } else {
+      setNotification(
+        NOTIFICATION_MAP_INTERNAL_BALANCES[TransactionStatus.WRITE_ERROR]
+      );
+      setTransactionStatus(TransactionStatus.AUTHORIZING);
+    }
```

Comment:
Can you elaborate on why we need to set the transaction status after erroring out?

Reply:
So the user can reject on metamask and still be able to try again https://www.loom.com/share/32d8aad314204ed3922d10062f4ba639?sid=46aefd58-a45b-4944-af3d-41e26e516747